libcaf_core/caf/config_option.hpp

```diff
@@ -54,10 +54,18 @@ class config_option {
   config_option(string_view category, string_view name, string_view description,
                 const meta_state* meta, void* value = nullptr);
 
+  config_option(const config_option&);
+
   config_option(config_option&&) = default;
 
+  config_option& operator=(config_option);
```

Comment:
Should read `operator=(const config_option&)`. For some types it makes sense to provide only this signature, but since we want to have a separate move-assign operator we should use the standard signatures.